### PR TITLE
Add support for ingesting/retrieving Seurat dimensional reduction data

### PR DESCRIPTION
**Pull request by aaronwolen**
Adds support for storing and retrieving dimensional reduction results from a Seurat object's `DimReduc` slot. The updated docs provide details about how the various results are stored on disk but here's what the directory structure looks like after converting `pbmc_small`:

```
/var/folders/rn/r3r2rnt56z98l344kf1tq5ph0000gn/T//RtmpcOhqoq/test-scdata
├── __tiledb_group.tdb
└── scgroup_RNA
    ├── X
    │   ├── __1646331732958_1646331732958_b7cbe19af0c14a6a91ce7458a4283f2d_11
    │   ├── __1646331732958_1646331732958_b7cbe19af0c14a6a91ce7458a4283f2d_11.ok
    │   ├── __meta
    │   └── __schema
    ├── __tiledb_group.tdb
    ├── obs
    │   ├── __1646331733046_1646331733046_628ab9c763ae4870b17c57b4baa2ca0d_11
    │   ├── __1646331733046_1646331733046_628ab9c763ae4870b17c57b4baa2ca0d_11.ok
    │   ├── __meta
    │   └── __schema
    ├── obsm
    │   ├── __tiledb_group.tdb
    │   ├── dimreduction_pca
    │   └── dimreduction_tsne
    ├── var
    │   ├── __1646331733004_1646331733004_69f59c9e516b47298bfeae951fa545f3_11
    │   ├── __1646331733004_1646331733004_69f59c9e516b47298bfeae951fa545f3_11.ok
    │   ├── __meta
    │   └── __schema
    └── varm
        ├── __tiledb_group.tdb
        └── dimreduction_pca
```

## Changes

- `SCDataset`'s `from_seurat()` method will now automatically detect the presence of dimensional reduction results
- `SCGroup` gains `from_seurat_dimreduction()` and `to_seurat_dimreduction()` methods that handle the conversions
- new `AnnotationMatrixGroup` class for managing the `obsm` and `varm` fields of an `SCGroup` and provide a method for adding new `AnnotationMatrix`s 
- `TiledbGroup` gains a method for listing object URIs and filtering results by a prefix

### CODE REVIEW

File: R/AnnotationMatrixGroup.R [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+#' Class for representing a TileDB group containing one or more
+#' [`AnnotationMatrix`] arrays that share a common dimension name.
+#' @export
+AnnotationMatrixGroup <- R6::R6Class(
+  classname = "AnnotationMatrixGroup",
+  inherit = TiledbGroup,
+
+  public = list(
+    #' @field arrays Named list of [`AnnotationMatrix`] arrays in the group
+    # TODO: Should this be named annotation_matrices? or annot_mats to avoid
```
Comment (johnkerl) [inline R/AnnotationMatrixGroup.R:13]: I prefer `AnnotationMatrixGroup`; makes it clear it's not simply an unrelated plural `AnnotationMatrices` but rather an intentionally colocated _group_.

File: R/AnnotationMatrixGroup.R [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    # having to pluralize matrix?
```
Comment (johnkerl) [inline R/AnnotationMatrixGroup.R:14]: I think we can always replace "`AnnotationMatrix`s" with "`AnnotationMatrix` objects" or somesuch

**Review by johnkerl — COMMENTED** [review]
Just a few light comments at this point -- I like this a lot! :)

File: R/AnnotationMatrixGroup.R (hunk 1 above)
Comment (aaronwolen) [reply R/AnnotationMatrixGroup.R:13]: Thanks for the feedback! I struggled with this one.

**Pull request merged**